File: MobileClient/app/src/main/java/ru/ex/mobileclient/models/PublicLink.kt
package ru.ex.mobileclient.models

import com.fasterxml.jackson.annotation.JsonIgnoreProperties

@kotlinx.serialization.Serializable
@JsonIgnoreProperties(ignoreUnknown = true)
data class PublicLink(
    val id: String,
    val isFile: Boolean,
    val contentID: Int,
    val isDownloadCountRestricted: Boolean,
    var downloadCount: Int
)

File: PcClient/Models/PublicLink.cs
namespace Server.Models
{
    public class PublicLink
    {
        public string ID { get; set; }

        public bool IsFile { get; set; }

        public int ContentID { get; set; }

        public bool IsDownloadCountRestricted { get; set; }

        public int DownloadCount { get; set; }
    }
}
